code-for-canada/project-thundercat
# [Inbox] Add email action handles action data

# Description

No user facing changes. This preps us to show a saved email action to the user.

- Followed EmivInboxRedux tech spec for storing data
- Re-named components to be consistent 
- Defined enums and PropType shapes in eMIB/constants.js
- Modified `addEmail` to take in an `actionShape` and added a test
- Saved state of the add email form into state and then redux on save button press

The last part of "Add email" is displaying it in the Email component.

## Type of change

- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor

## Screenshot

No changes to the UI.

# Testing

Manual steps to reproduce this functionality:

1.  Navigate to eMIB Sample Test Inbox.
2.  Click on "Add email" or "Add task" and notice no breaking changes.

# Checklist

~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)~~
~~- [ ] My changes look good on IE 10+, Firefox, and Chrome~~
